### PR TITLE
adapt pytorch-ie template to hydra v1.2

### PR DESCRIPTION
**Pull request by Bhuvanesh-Verma**
This PR adapts the pytorch-ie hydra template to hydra 1.2 so that the working directory is no longer changed by default. These changes are based on the similar [PR](https://github.com/ashleve/lightning-hydra-template/pull/313) from original lightening hydra template.

### CODE REVIEW

File: configs/predict.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ defaults:
   - _self_
   - dataset: conll2003
   - serializer: json
```
Comment (ArneBinder) [inline configs/predict.yaml:7]: did we miss to add `hydra` here?

File: configs/predict.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@ defaults:
...
+  - paths: default.yaml
 
   # enable color logging
   - override hydra/hydra_logging: colorlog
```
Comment (ArneBinder) [inline configs/predict.yaml:11]: can we add `task_name: "predict"` down below to follow the scheme of train/eval?

File: configs/test.yaml [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
+ignore_warnings: False
 
 print_config: True
 
-ignore_warnings: True
+# passing checkpoint path is necessary for evaluation
+ckpt_path: null
 
-seed: null
+original_work_dir: ${hydra:runtime.cwd}
```
Comment (ArneBinder) [inline configs/test.yaml:31]: is this needed anymore?

File: configs/predict.yaml (hunk 2 above)
Comment (ArneBinder) [reply configs/predict.yaml:11]: and also `data_dir` is already defined in `paths`

File: configs/test.yaml [hunk 4]
```diff
@@ -1,32 +1,38 @@
...
 
-name: "default"
+data_dir: ${original_work_dir}/data/
```
Comment (ArneBinder) [inline configs/test.yaml:33]: already defined in `paths`

File: configs/train.yaml [hunk 5]
```diff
@@ -26,37 +28,45 @@ defaults:
...
-data_dir: ${original_work_dir}/data/
+# default name for the experiment, determines output directory path
+# (you can overwrite this in experiment configs)
+name: "default"
 
-# pretty print config at the start of the run using Rich library
-print_config: True
+# tags to help you identify your experiments
+# (you can overwrite this in experiment configs)
+tags: ["mnist"]
```
Comment (ArneBinder) [inline configs/train.yaml:43]: this tag is wrong

File: configs/train.yaml [hunk 6]
```diff
@@ -26,37 +28,45 @@ defaults:
...
+# pretty print config at the start of the run using Rich library
+print_config: True
 
 # set False to skip model training
 train: True
-
 # evaluate on test set, using best model weights achieved during training
 # lightning chooses best weights based on the metric specified in checkpoint callback
-test: False
+test: True
```
Comment (ArneBinder) [inline configs/train.yaml:58]: why is this changed?

File: configs/train.yaml [hunk 7]
```diff
@@ -26,37 +28,45 @@ defaults:
...
-# default name for the experiment, determines logging folder path and trained model location
-# (you can overwrite this name in experiment configs)
-name: "default"
+# path to original working directory
+# hydra hijacks working directory by changing it to the new log directory
+# https://hydra.cc/docs/next/tutorials/basic/running_your_app/working_directory
+original_work_dir: ${hydra:runtime.cwd}
+
+# path to folder with data
+data_dir: ${original_work_dir}/data/
```
Comment (ArneBinder) [inline configs/train.yaml:69]: as above

File: configs/trainer/default.yaml [hunk 8]
```diff
@@ -1,5 +1,8 @@
 _target_: pytorch_lightning.Trainer
 
+default_root_dir: ${paths.output_dir}
+enable_model_summary: False
```
Comment (ArneBinder) [inline configs/trainer/default.yaml:4]: what is `enable_model_summary`? why is this added within this PR?

**Review by ArneBinder — CHANGES_REQUESTED** [review]
Thanks for the PR, some comments below. Also, please change the minimal hydra version in requirements.txt to `1.2`. Can you link to the respective PR of the original template in the PR description?

File: configs/trainer/default.yaml (hunk 8 above)
Comment (Bhuvanesh-Verma) [reply configs/trainer/default.yaml:4]: This was a part of the PR, but maybe not required wrt current changes.

File: configs/train.yaml (hunk 6 above)
Comment (Bhuvanesh-Verma) [reply configs/train.yaml:58]: Yeah, I guess not required.

File: configs/train.yaml (hunk 5 above)
Comment (Bhuvanesh-Verma) [reply configs/train.yaml:43]: Should we remove this attribute or maybe change to something like `conll2003_train` ?

File: configs/test.yaml (hunk 3 above)
Comment (Bhuvanesh-Verma) [reply configs/test.yaml:31]: No, will remove it.

File: configs/predict.yaml (hunk 2 above)
Comment (Bhuvanesh-Verma) [reply configs/predict.yaml:11]: Ah yeah right, task_name is required for logs so must be added.

File: configs/predict.yaml (hunk 1 above)
Comment (Bhuvanesh-Verma) [reply configs/predict.yaml:7]: I guess predict was little half baked since there was nothing to compare it with from original repo. However, I have updated it now according to train and test configs.

File: configs/train.yaml (hunk 5 above)
Comment (ArneBinder) [reply configs/train.yaml:43]: just "conll2003"

Comment (ArneBinder) [reply configs/train.yaml:43]: or maybe even better: "dataset=conll2003"?

File: configs/trainer/default.yaml (hunk 8 above)
Comment (ArneBinder) [reply configs/trainer/default.yaml:4]: please have a look what this parameter does and then decide, if it makes sense to include it.

File: configs/train.yaml (hunk 5 above)
Comment (Bhuvanesh-Verma) [reply configs/train.yaml:43]: Yeah exactly, that serves the purpose of attribute `tags`. Thanks, I'll update it.

File: configs/trainer/default.yaml (hunk 8 above)
Comment (Bhuvanesh-Verma) [reply configs/trainer/default.yaml:4]: This parameter was eventually removed in https://github.com/ashleve/lightning-hydra-template/pull/358 . Not sure why it was even introduced at first place.

File: README.md [hunk 9]
```diff
@@ -374,10 +374,10 @@ python train.py -m datamodule.batch_size=32,64,128 model.lr=0.001,0.0005
 ```bash
 # this will run hyperparameter search defined in `configs/hparams_search/mnist_optuna.yaml`
```
Comment (ArneBinder) [inline README.md:375]:
```suggestion
# this will run hyperparameter search defined in `configs/hparams_search/conll2003_optuna.yaml`
```

File: configs/trainer/default.yaml (hunk 8 above)
Comment (ArneBinder) [reply configs/trainer/default.yaml:4]: ok, then let's remove it directly

File: configs/predict.yaml [hunk 10]
```diff
@@ -5,14 +5,14 @@ defaults:
   - _self_
   - dataset: conll2003
   - serializer: json
+  - hydra: default.yaml
+  - paths: default.yaml
 
   # enable color logging
   - override hydra/hydra_logging: colorlog
   - override hydra/job_logging: colorlog
```
Comment (ArneBinder) [inline configs/predict.yaml:13]: is this already defined in `hydra/defaults.yaml`, isn't it?

File: configs/test.yaml [hunk 11]
```diff
@@ -1,33 +1,37 @@
...
   - trainer: default.yaml
-  - log_dir: evaluation.yaml
-
+  - hydra: default.yaml
+  - paths: default.yaml
   - experiment: null
 
   # enable color logging
   - override hydra/hydra_logging: colorlog
   - override hydra/job_logging: colorlog
```
Comment (ArneBinder) [inline configs/test.yaml:16]: as above, is this already defined in `hydra/defaults.yaml`, isn't it?

**Review by ArneBinder — CHANGES_REQUESTED** [review]
some minors. But I'm not so much satisfied with the `task_name` parameter because it clashes with our taskmodules etc. What do you think about `pipeline_type`? Maybe with values `training`, `evaluation`, and `prediction`?

File: configs/test.yaml (hunk 11 above)
Comment (Bhuvanesh-Verma) [reply configs/test.yaml]: fixed in [e884f23](https://github.com/ChristophAlt/pytorch-ie-hydra-template/pull/42/commits/e884f233ad799e3bbebd68b922e6570403f7b773)

File: configs/predict.yaml (hunk 10 above)
Comment (Bhuvanesh-Verma) [reply configs/predict.yaml]: fixed in [e884f23](https://github.com/ChristophAlt/pytorch-ie-hydra-template/pull/42/commits/e884f233ad799e3bbebd68b922e6570403f7b773)

File: README.md (hunk 9 above)
Comment (Bhuvanesh-Verma) [reply README.md]: fixed in [7ab0710](https://github.com/ChristophAlt/pytorch-ie-hydra-template/pull/42/commits/7ab0710af544c1c9b38236a7aff0f9badb2c0a0a)

**Review by ArneBinder — APPROVED** [review]
